MSAL/src/native_auth/public/state_machine/error/ResetPasswordStartError.swift

```diff
@@ -24,41 +24,44 @@
 
 import Foundation
 
-@objc
+@objcMembers
 public class ResetPasswordStartError: MSALNativeAuthError {
-    /// An error type indicating the type of error that occurred
-    @objc public let type: ResetPasswordStartErrorType
+    let type: ResetPasswordStartErrorType
 
     init(type: ResetPasswordStartErrorType, message: String? = nil) {
         self.type = type
-        super.init(message: message)
+        super.init(identifier: type.rawValue, message: message)
     }
 
+    /// Describes an error that provides messages describing why an error occurred and provides more information about the error.
     public override var errorDescription: String? {
-        if let description = super.errorDescription {
-            return description
-        }
+        return super.errorDescription ?? type.rawValue
+    }
+
+    /// Returns `true` if the error requires to use a browser.
+    public var isBrowserRequired: Bool {
+        return type == .browserRequired
+    }
+
+    public var isUserDoesNotHavePassword: Bool {
+        return type == .userDoesNotHavePassword
+    }
+
+    /// Returns `true` if the user that is trying to sign in cannot be found.
+    public var isUserNotFound: Bool {
+        return type == .userNotFound
+    }
 
-        switch type {
-        case .browserRequired:
-            return "Browser required"
-        case .generalError:
-            return "General error"
-        case .userDoesNotHavePassword:
-            return "User does not have a password"
-        case .userNotFound:
-            return "User not found"
-        case .invalidUsername:
-            return "Invalid username"
-        }
+    /// Returns `true` when the username introduced is not valid.
+    public var isInvalidUsername: Bool {
+        return type == .invalidUsername
     }
 }
 
-@objc
-public enum ResetPasswordStartErrorType: Int {
-    case browserRequired
-    case generalError
-    case userDoesNotHavePassword
-    case userNotFound
-    case invalidUsername
+public enum ResetPasswordStartErrorType: String, CaseIterable {
```

Comment:
I suppose you didn't make this enum available to objective C, because we need to use Int as rawValue, right? We could use an additional attribute in that case, so we can use it from Objective-C and Swift

Reply:
For objective-c developers, they can do:

```objective-c
- (void)onSignInPasswordErrorWithError:(SignInPasswordStartError * _Nonnull)error {
  error.identifier
  error.errorDescription
  error.isBrowserRequired
  error.is...
}
```
Right now `identifier` is a String, so they can't use it to compare with `SignInPasswordStartErrorType`.

If we change `identifier` to be an Int, I think it would work. We can also rename it as `errorCode` if you like. But in this case it would be better to assign proper error codes like the ones present in `MSALError.h`

Reply:
I agree on making the "identifier" variable an integer, so we could change all the error type enum (example: SignUpStartErrorType) rawValue to integer, so they can be exposed to ObjectiveC. I don't agree to rename the identifier variable to something else.

Reply:
Updated as discussed. I've updated the PR description to reflect these changes, too.